pandas/tests/indexes/datetimes/test_ops.py

```diff
@@ -144,6 +144,38 @@ def test_numpy_minmax(self):
             tm.assert_raises_regex(
                 ValueError, errmsg, np.argmax, dr, out=0)
 
+    @pytest.mark.parametrize('test_input, rounder, freq, expected', [
+        # GH 14440 & 15578
+        ('2016-10-17 12:00:00.0015', 'round', 'ms',
+         '2016-10-17 12:00:00.002000'),
+        ('2016-10-17 12:00:00.0015', 'round', 'us',
+         '2016-10-17 12:00:00.0015'),
+        ('2016-10-17 12:00:00.0015', 'round', 'ns',
+         '2016-10-17 12:00:00.0015'),
+        ('2016-10-17 12:00:00.00149', 'round', 'ms',
+         '2016-10-17 12:00:00.001000'),
+        ('2016-10-17 12:00:00.001501031', 'round', '10ns',
+         '2016-10-17 12:00:00.001501030'),
+        # GH 19206 - times far in the future and past rounding incorrectly
+        ('2117-01-01 00:00:45', 'floor', '15s', '2117-01-01 00:00:45'),
+        ('2117-01-01 00:00:45', 'ceil', '15s', '2117-01-01 00:00:45'),
+        ('2117-01-01 00:00:45.000000012', 'floor', '10ns',
+         '2117-01-01 00:00:45.000000010'),
+        ('1823-01-01 00:00:01', 'floor', '1s', '1823-01-01 00:00:01'),
+        ('1823-01-01 00:00:01', 'ceil', '1s', '1823-01-01 00:00:01'),
+        ('1823-01-01 00:00:01.000000012', 'floor', '10ns',
+         '1823-01-01 00:00:01.000000010'),
+        ('1823-01-01 00:00:01.000000012', 'ceil', '10ns',
+         '1823-01-01 00:00:01.000000020')
+    ])
+    def test_round_ops(self, test_input, rounder, freq, expected):
+        for tz in self.tz:
```

Comment:
I am not sure why tz was not made into a fixture, but I guess ok here

Reply:
I'll take care of that too. I don't know why either. Must have overlooked it.

Reply:
thanks!